src/Components/DetailItem/DropDownBox.js

```diff
@@ -0,0 +1,174 @@
+import React, { Component } from 'react';
+import Select from 'react-select';
+import OrderBox from './OrderBox';
+import { COLOR, SIZE } from './ITEM';
+import './DropDownBox.scss';
+
+class DropDownBox extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      number: 0,
+      value: '',
+      size: '',
+      amount: 1,
+      selectPrice: 16200,
+      defaultPrice: 0,
+      click: false,
+      click2: false,
+      item: [],
+      orderBox: [],
+      priceBox: 0,
+      heartChange: true,
+    };
+  }
+
+  componentDidUpdate(prevprops, prevstate) {
+    const { orderBox } = this.state;
+    if (prevstate.orderBox !== this.state.orderBox) {
+      const priceBox = orderBox.reduce((a, b) => {
+        a = a + b.priceBox * b.amount;
+        return a;
+      }, 0);
+      this.setState({ priceBox });
+    }
+  }
+
+  // 상품 클릭시 클릭 바뀌게
+  handleChange = (e) => {
+    const { number } = this.state;
+    this.setState({ number: number + 1, value: e.label });
+  };
+
+  //사이즈 클릭시 값이 누적되고 오더박스 카드가 생김,카드 종류별로 담아주는 기능 구현
+  handleSizeCheck = (e) => {
+    const { click2, selectPrice, item, value, orderBox, id2 } = this.state;
+    let arr = [...item];
+    arr.push(selectPrice);
+    //중복된 배열이 있을때 걸러주는 기능 추가
+    let orderBox2 = [...orderBox];
+    let orderBox3 = orderBox2.filter((el) => (el.value === value && el.size === e.label ? el : ''));
```

Comment:
filter는 return값의 Boolean을 비교해서 true면 해당 요소를 그대로 반환합니다 그래서 아래와 같이 적어주시면 됩니다.
```suggestion
    let orderBox3 = orderBox2.filter((el) => (el.value === value && el.size === e.label));
```

Reply:
좋은정보 감사합니다 바로 반영하였습니다!